modules/nw-ovn-k-baseline-adminnetwork-policy.adoc

```diff
@@ -0,0 +1,141 @@
+
+`BaselineAdminNetworkPolicy` (BANP) is a cluster-scoped custom resource definitions (CRDs). As a {product-title} administrator, you can use BANP to setup and enforce optional baseline network policies that are overridable by users using `NetworkPolicies` if need be. Rule actions for BANP are `allow` or `deny`.
+
+BANP allows administrators to specify:
+
+* A `subject` that consists of a set of pods that is selected by the policy.
+
+* A list of ingress rules to be applied for all ingress traffic towards the `subject`.
+
+* A list of egress rules to be applied for all egress traffic from the `subject`.
+
+[NOTE]
+====
+`BaselineAdminNetworkPolicy` is a `TechnologyPreviewNoUpgrade` feature that can be enabled on test clusters that are not in production. The feature is behind the flag `enable-admin-network-policy`.
+====
+
+[discrete]
+[id="BaselineAdminNetworkPolicy-example"]
+== BaselineAdminNetworkPolicy example
+
+The `BaselineAdminNetworkPolicy` resource is a cluster singleton `object` that can be used as a guardrail policy incase a passed traffic policy does not match any `NetworkPolicies` in the cluster. BANP can also be used as a default security model that provides guardrails that intra-cluster traffic is blocked by default and user will need to use `NetworkPolicy` to allow known traffic. You must use `default` as the name when creating a `BaselineAdminNetworkPolicy` object.
+
+
+.Example YAML file for BANP
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: BANP_example <1>
+spec:
+  subject:
+    namespaces:
+      matchLabels:
+          kubernetes.io/metadata.name: gryffindor
+  ingress:<2>
+  - name: "deny-all-ingress-from-slytherin"
+    action: "Deny"
+    from:
+    - pods:
+        namespaces:
+          namespaceSelector:
+            matchLabels:
+              conformance-house: slytherin
+        podSelector:
+          matchLabels:
+            conformance-house: slytherin
+  egress:<2>
+  - name: "pass-all-egress-to-slytherin"
+    action: "Allow" <3>
+    to:
+    - pods:
+        namespaces:
+          namespaceSelector:
+            matchLabels:
+              conformance-house: slytherin
+        podSelector:
+          matchLabels:
+            conformance-house: slytherin
+----
+<1> Policy name
+<2> BANP have both ingress and egress rules.
+<3> BANP `action` rules for ingress and egress only accepts values of `Deny` and `Allow`.
+====
+
+
+[discrete]
+[id="BaselineAdminNetworkPolicy-default-deny-example"_{context}]
+== BaselineAdminNetworkPolicy Deny example
+The following BANP singleton ensures that the administrator has set up a default deny policy for all ingress monitoring traffic coming into the tenants at `internal` security level. When combined with the "AdminNetworkPolicy Pass example", this deny policy acts as a guardrail policy for all ingress traffic that is passed by the ANP `pass-monitoring` policy.
```

Comment:
I don't have a suggestion really, but if there's another word that could be used in place of "guardrail", might be worth switching.

Edit: Seems that that would involve changing quite a few instances, so probably fine as-is, but something to ponder.

Reply:
Yeah, I don't disagree. I just couldn't think of suitable alternative. I'll keep thinking about it though.